Adds code for #7 to ingest transcript data from .srt files.

takposha opened the pull request:
Fixes #7

takposha committed:
Adds code for #7 to ingest transcript data from .srt files.

pushyamig commented:
I will review this in the afternoon.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,233 @@
+
+from dotenv import load_dotenv
+
+load_dotenv()
```
This needs to be surrounded by `try-catch` and should appropriate log message denoting the error case.We have to assume that who ever is running did not attempt to setup the `.env` file

takposha replied:
That is a good one to catch, never thought of that. 
In the next PR, I added a way to check that empty values are not being passed for the API parameters. 

But I will also add in an explicit check for the .env file as well.

pushyamig replied:
when a suggestion is made to the PR, usually we tend to resolve in the same PR. if the suggesting is complicated or making it a big Change to the existing PR then we create an issue and say we will resolve in that issue. It's up to you if you want to resolve the issue here or next

takposha replied:
Got it. Let me go back to this branch and work on this in this PR chain itself then.

takposha replied:
Commit [7cc642f](https://github.com/tl-its-umich-edu/annoto-gai/pull/8/commits/7cc642fc9d35f9ebc7960c6d21f509086e95a198) addresses this comment,

pushyamig commented on captionsProcessor.ipynb:
`from helper import *` is not a best practice. Just bring the functions you needed rather do a `*`

takposha replied:
Commit [7cc642f](https://github.com/tl-its-umich-edu/annoto-gai/pull/8/commits/7cc642fc9d35f9ebc7960c6d21f509086e95a198) addresses this comment,

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,233 @@
+import os
+import glob
+import sys
+import pickle
```
Optional: (Nitpick), remove the import that are not used i the file. I understand the next PR seems to use this PR.

takposha replied:
I will keep that in mind moving forward. The next PR's code will use the pickle library and got left behind here instead of being removed.

takposha committed:
- Add initial setup instructions to README.md
- Update captionsProcessor.ipynb to import necessary functions from helper.py
- Add error handling for missing .env file in Config class

takposha committed:
helper.py - Moved load_dotenv to the config intialization
captionsProcessor.ipynb - Add reload extension
requirements.txt - Added IPython requirement

pushyamig reviewed (approved):
This is Looking Good! Thanks for accommodating my request with Github Workflow of Forking instead of creating the branch.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+def getCombinedTranscripts(
+    videoNames,
+    captionsFolder="Captions",
+    windowSize=30,
```
Usually the `windowSize` can also be configurable option.

takposha replied:
Yes, my next PR will be focusing on setting up a better configuration Class and handling errors. Question generation will be after that so no issues build up.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+
+    Returns:
+        dict or str: If only one video name is provided, returns the combined transcript as a string.
+                    If multiple video names are provided, returns a dictionary of combined transcripts for each video.
```
I like the way you are using docstring, I think you could consider using Python Typing when you feel convenient  that way you don't need to mention return types, for a functions. https://docs.python.org/3/library/typing.html

But I won't stress on it now since the code is build up and it might impede the progress.

takposha replied:
Hmm, I could look into Python Typing too!

I'm using this format mostly because it's what Copilot generates for me. I take whatever it generates and edit it as needed to match up my requirements.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+                    If multiple video names are provided, returns a dictionary of combined transcripts for each video.
+    """
+
+    if type(videoNames) == str:
```
The check here should be made if the `videoNames` is empty string and if it is then log and exit the process. Since without the videoname it doesn't make sense to proceed further

takposha replied:
Yes, my next PR will be focusing on setting up a better configuration Class and handling errors. Question generation will be after that so no issues build up.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+
+    srtFiles, transcripts, sentences, combinedTranscripts = {}, {}, {}, {}
+    for video in videoNames:
+        srtFiles[video] = glob.glob(f"{captionsFolder}/{video}/*.srt")
```
Surround this with try/catch since the error here could be file not found if the user  did not configured the file. Again, log/exit the process

takposha replied:
Will do.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+    if type(srtFile) == list:
+        srtFile = srtFile[0]
+
+    with open(srtFile, "r") as f:
```
This could cause a file not found error. so need some action to handle the situation. So the code should be know how to handle the happy and unhappy path

takposha replied:
Noted. This will be handled in a future PR, while handling #11.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+        line = line.strip()
+        if line.isdigit():
+            continue
+        elif "-->" in line:
```
Optional: make `-->` as constant. when ever a string is repeated it should be made a constant. So that you don't need to retype multiple times

takposha replied:
Noted. This will be handled in a future PR, while handling #11.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+        elif "-->" in line:
+            startTime, endTime = line.split("-->")
+            startTime = datetime.strptime(startTime.strip(), "%H:%M:%S,%f")  # .time()
+            endTime = datetime.strptime(endTime.strip(), "%H:%M:%S,%f")  # .time()
```
Optional: make `"%H:%M:%S,%f"` as constant. when ever a string is repeated it should be made a constant. So that you don't need to retype multiple times

takposha replied:
Noted. This will be handled in a future PR, while handling #11.

pushyamig commented on helper.py:
```diff
@@ -0,0 +1,244 @@
+            )
+            sentence = ""
+
+    return pd.DataFrame(transcript)
```
I would make this a variable and then return the variable that hold the `df`. good for debugging purposes. also log what is returned as debug

```
transcript_df = pd.DataFrame(transcript)
return transcript_df
```

pushyamig replied:
I would log something like `transcript_df.head(5)`

takposha replied:
Noted. This will be handled in a future PR, while handling #11.

takposha replied:
I will also extend this to the topics generated and such in the logging so we can track any issues there as well.
